src/ParquetMsg.chpl

```diff
@@ -820,45 +824,76 @@ module ParquetMsg {
           tagData = false; // turn off so we only run once
         }
 
+        var whereNull = makeDistArray(len, bool);
         // Only integer is implemented for now, do nothing if the Parquet
         // file has a different type
         if ty == ArrowTypes.int64 || ty == ArrowTypes.int32 {
           var entryVal = createSymEntry(len, int);
-          readFilesByName(entryVal.a, filenames, sizes, dsetname, ty);
+          readFilesByName(entryVal.a, whereNull, filenames, sizes, dsetname, ty, hasNonFloatNulls=hasNonFloatNulls);
           var valName = st.nextName();
-          st.addEntry(valName, entryVal);
+          if hasNonFloatNulls && (|| reduce whereNull) {
+            // if we have non-float nulls and there's at least one null
+            var floatEntry = createSymEntry(len, real);
+            floatEntry.a = (entryVal.a):real;
+            ref fa = floatEntry.a;
+            [(t, f) in zip(whereNull, fa)] if t then f = nan;
```

Comment:
I am confused by this, aren't we setting the null values in the C++ code?

Reply:
ah yah we do but since they're not float arrays, it ends up being `0`s. I should just remove that line setting it to nan in c++ code cause it isn't doing anything and it's just confusing